docs/validate-webhook.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: tekton.dev/v1alpha1
```

Comment:
Instead of create the custom image with the custom executable, would you use a task something like this?
I haven't debugged  this but it should work with  only minor changes if necessary.

```
apiVersion: tekton.dev/v1alpha1
kind: Task
metadata:
  name: validate-webhook
spec:
  inputs:
    params:
      - name: Payload
        description: Payload of Event Received
      - name: X-Hub-Signature
        description: Hash of the Request Received
      - name: Github-Secret
        description: Secret used to configure webhook
  steps:
    - name: validate
      image: python
      command: ["/bin/bash"]
      args:
      - -ce
      - |
        set -e
        cat <<EOF | python
        import hashlib, os, hmac, six
        secret = os.environ.get('GithubSecret') 
        payload = $(inputs.params.Payload)
        signature = $(inputs.params.X-Hub-Signature)
        expected = hmac.new(secret, payload, hashlib.sha1).hexdigest()
        if expected is not None:
          if not isinstance(expected, six.text_type):
            expected = six.text_type(expected)
          sig_parts = signature.split("=", 1)
          if not isinstance(sig_parts[1], six.text_type):
            sig_parts1 = six.text_type(sig_parts[1])
          else:
            sig_parts1 = sig_parts[1]
          if len(sig_parts) > 1 and sig_parts[0] == "sha1" and hmac.compare_digest(sig_parts1, expected):
            exit(0)
        exit(1)
        EOF 
      env:
        - name: GithubSecret
          valueFrom:
           secretKeyRef:
             name: $(inputs.params.Github-Secret)

```

Reply:
I created this task with some minor changes and tested it. It works.